go/roothash/api/api.go

```diff
@@ -427,6 +425,8 @@ func (e *ExecutorCommittedEvent) EventKind() string {
 
 // ExecutionDiscrepancyDetectedEvent is an execute discrepancy detected event.
 type ExecutionDiscrepancyDetectedEvent struct {
+	// Rank is the rank of the transaction scheduler.
+	Rank uint64 `json:"rank"`
```

Comment:
No. When discrepancy is detected, the backup workers need to know the rank (or ID) of the proposer so they can fetch the right proposal, execute txs and submit a commitment. Since they are listening to events, this is the way to go. If we remove this field, they would need to query the consensus.

Reply:
oh ok thanks